src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1648,10 +1648,21 @@ internal static CollectionExpressionTypeKind GetCollectionExpressionTypeKind(CSh
             {
                 return CollectionExpressionTypeKind.CollectionBuilder;
             }
-            else if (implementsIEnumerable(compilation, destination))
+            else if (implementsSpecialInterface(compilation, destination, SpecialType.System_Collections_Generic_IEnumerable_T))
             {
                 elementType = null;
-                return CollectionExpressionTypeKind.CollectionInitializer;
+                return CollectionExpressionTypeKind.GenericIEnumerable;
+            }
+            else if (implementsSpecialInterface(compilation, destination, SpecialType.System_Collections_IEnumerable))
+            {
+                // ^ This implementation differs from Binder.CollectionInitializerTypeImplementsIEnumerable().
```

Comment:
>^  [](http://example.com/codeflow?start=19&length=2)

Is this intentional?

Reply:
Yes, this comment applies to `implementsSpecialInterface` call above. This still seems the best place for this large comment...